frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/install-lso-sc.tsx

```diff
@@ -0,0 +1,152 @@
+
+  return Promise.all(promises).then(() => k8sCreate(OCSServiceModel, ocsObj));
+};
+
+export const CreateOCS = withHandlePromise<CreateOCSProps & HandlePromiseProps>((props) => {
+  const {
+    handlePromise,
+    errorMessage,
+    inProgress,
+    match: {
+      params: { appName, ns },
+    },
+  } = props;
+  const [filteredNodes, setFilteredNodes] = React.useState<string[]>([]);
+  const [storageClass, setStorageClass] = React.useState<StorageClassResourceKind>(null);
+  const [nodes, setNodes] = React.useState<NodeKind[]>([]);
+  const [lvsList, setLvsList] = React.useState<K8sResourceKind[]>([]);
+  const dispatch = useDispatch();
+  const [lvsData, lvsLoaded, lvsLoadError] = useK8sWatchResource<K8sResourceKind[]>(lvsResource);
+
+  const handleStorageClass = (sc: StorageClassResourceKind) => {
+    setStorageClass(sc);
+
+    if (sc) {
+      const [lvs] = lvsList.filter((l) => l?.spec?.storageClassName === sc?.metadata?.name);
+      setFilteredNodes(
+        lvs?.spec?.nodeSelector?.nodeSelectorTerms?.[0]?.matchExpressions?.[0]?.values,
+      );
+    } else {
+      setFilteredNodes([]);
+    }
+  };
+
+  React.useEffect(() => {
+    if ((lvsLoadError || lvsData.length === 0) && lvsLoaded) {
+      setLvsList([]);
+      setFilteredNodes([]);
+    } else if (lvsLoaded) {
+      setLvsList(lvsData);
+    }
+  }, [lvsData, lvsLoaded, lvsLoadError]);
+
+  const submit = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    // eslint-disable-next-line promise/catch-or-return
+    handlePromise(makeOCSRequest(nodes, storageClass)).then(() => {
+      dispatch(setFlag(OCS_CONVERGED_FLAG, true));
+      dispatch(setFlag(OCS_FLAG, true));
+      history.push(
+        `/k8s/ns/${ns}/clusterserviceversions/${appName}/${referenceForModel(
+          OCSServiceModel,
+        )}/${getName(getOCSRequestData())}`,
+      );
+    });
+  };
+
+  const onlyNoProvSC = React.useCallback(
+    (sc: StorageClassResourceKind) => sc?.provisioner === NO_PROVISIONER,
+    [],
+  );
+
+  return (
+    <div className="co-m-pane__form">
+      <OCSAlert />
+      <Form className="co-m-pane__body-group">
+        <StorageClassSection handleStorageClass={handleStorageClass} filterSC={onlyNoProvSC}>
+          <PVsAvailableCapacity
+            replica={getOCSRequestData().spec.storageDeviceSets[0].replica}
+            data-test-id="ceph-ocs-install-pvs-available-capacity"
+            sc={storageClass}
+          />
+        </StorageClassSection>
+        <h3 className="co-m-pane__heading co-m-pane__heading--baseline ceph-ocs-install__pane--margin">
+          <div className="co-m-pane__name">Nodes</div>
+        </h3>
+        {storageClass ? (
+          <SelectNodesSection
+            table={AttachedDevicesNodeTable}
+            customData={{ filteredNodes, nodes, setNodes }}
+          />
+        ) : (
+          <div className="ceph-ocs-install__no--nodes--text">No nodes available to show</div>
```

Comment:
Its not anything about matching criteria. It displayed because no SC is selected. @yuvalgalanti can help us with this and can be taken in a follow up PR.